torch/_functorch/_aot_autograd/collect_metadata_analysis.py

```diff
@@ -424,6 +438,20 @@ def inner(*flat_args):
                 else:
                     output_type = OutputType.alias_of_input
 
+                    # If the tensors are instances of FunctionalTensor, then record how many ViewMeta
+                    # instances the input has. That will be the base offset.
+                    f_arg = flat_f_args[base_idx]
+                    if isinstance(o, FunctionalTensor) and isinstance(
+                        f_arg, FunctionalTensor
+                    ):
+                        base_functional_offset = torch._functionalize_view_metas_size(
```

Comment:
Per my above comment - is this ever not zero?

Reply:
Yes. As you pointed out above, when we have in-place view functions.